model/src/main/kotlin/config/PackageSnippetChoice.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2023 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model.config
+
+import org.ossreviewtoolkit.model.RepositoryProvenance
+
+data class PackageSnippetChoice(
+    /**
+     * The provenance this snippet choice applies to.
+     */
+    val provenance: Provenance,
+
+    /**
+     * The snippet choices for this package.
+     */
+    val choices: List<SnippetChoice>
+)
+
+/**
+ * The URL of the [RepositoryProvenance] the snippet choice applies to.
+ */
+data class Provenance(val url: String)
```

Comment:
Since this `data class` has only a single property, could it be an [inline `value class`](https://kotlinlang.org/docs/inline-classes.html)?

Reply:
Unfortunately it causes a Jackson Exception on deserialization:
```
com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `org.ossreviewtoolkit.model.config.SnippetChoices` (no Creators, like default constructor, exist): cannot deserialize from Object value (no delegate- or property-based Creator)
 at [Source: (StringReader); line: 54, column: 6] (through reference chain: org.ossreviewtoolkit.model.config.RepositoryConfiguration["snippet_choices"]->java.util.ArrayList[0])
```